### PR TITLE
Implemented the Message Queue Server in Java

### PR DESCRIPTION
**Pull request by yadavpa1**

### CODE REVIEW

File: src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    /**
+     * Creates a new ledger for a topic partition.
+     *
+     * @param topic     The topic name.
+     * @param partition The partition number.
+     * @throws Exception If an error occurs while creating the ledger.
+     */
+    public void createLedger(String topic, int partition) throws Exception {
+        try (LedgerHandle ledger = bookKeeper.createLedger(
```
Comment (catchashu10) [inline src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java:33]: Why are we not using `topic `and `partition `to `createLedger `here?

File: src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * Writes a message to the specified topic partition ledger.
+     *
+     * @param topic     The topic name.
+     * @param partition The partition number.
+     * @param message   The message to write.
+     * @throws Exception If an error occurs while writing the message.
+     */
+    public void writeMessage(String topic, int partition, Message message) throws Exception {
+        try (LedgerHandle ledger = bookKeeper.openLedger(
+                generateLedgerId(topic, partition),
```
Comment (catchashu10) [inline src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java]: How can we use custom genereateLedgerId() if creating a ledger is not using a `topic `and `partition `to create the id.

**Comment by yadavpa1** [comment]
It is fixed.

File: src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java (hunk 1 above)
Comment (yadavpa1) [reply src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java:33]: Resolved in commit adf681753297ff834cea8db332359d743fd8bc1a "Resolved a minor bug in BookKeeperClient"

File: src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java (hunk 2 above)
Comment (yadavpa1) [reply src/main/java/com/clustercrew/messagequeue/BookKeeperClient.java]: Resolved in commit https://github.com/yadavpa1/DistributedMessageQueue/commit/adf681753297ff834cea8db332359d743fd8bc1a "Resolved a minor bug in BookKeeperClient"

**Review by Pratik-Sanghavi — APPROVED** [review]
Looks good to me

**Pull request merged**